fallback to download ffprobe if missing info

aeh opened the pull request:
this should fix these errors...

```
Error: ffprobe exited with code 1
...
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc02000be00] stream 0, offset 0x30: partial file
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc02000be00] Could not find codec parameters for stream 0
  (Video: h264 (avc1 / 0x31637661), none, 1280x720, 6418 kb/s): unspecified pixel format
Consider increasing the value for the 'analyzeduration' and 'probesize' options
```

aeh committed:
fallback to download ffprobe if missing info

aeh committed:
update docs

aeh committed:
2.0.0-dl.0

timlesallen commented:
@aeh Is this required for this card or is it a bonus inclusion?

timlesallen commented:
code climate check is not running on this PR - need to hook up.

timlesallen commented on src/index.js:
```diff
@@ -11,30 +13,53 @@ ffmpeg.setFfprobePath(ffprobe.path);
-        .pipe(buffer);
+module.exports = async function probe (url, opts = {}) {
+  const stream = got.stream(url, opts);
+  const input = opts.download ? await downloadStream(url, stream) : stream;
```
I think you may need to document the `download` option?

timlesallen commented on src/index.js:
```diff
@@ -11,30 +13,53 @@ ffmpeg.setFfprobePath(ffprobe.path);
+ * @param {stream} stream
+ * @returns {string} filename
+ */
+async function downloadStream (url, stream) {
```
@aeh I already wrote a module that might do exactly what we need here... **https://github.com/noblesamurai/retriable-download**

Could save some LOC and keep maintenance lower?

aeh committed:
use retriable-download, add doc for download opt

aeh committed:
reject properly on input errors

Note: the fluent-ffmpeg ffprobe command doesn't return anything so there
is no actual way to kill the process on an input error.
https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/issues/607

aeh committed:
2.0.0-dl.1
